src/compartment_sets.cpp

```diff
@@ -0,0 +1,670 @@
+
+
+#include "../extlib/filesystem.hpp"
+
+#include "utils.h"  // readFile
+
+#include <unordered_set>
+
+#include <bbp/sonata/compartment_sets.h>
+namespace bbp {
+namespace sonata {
+
+namespace fs = ghc::filesystem;
+
+namespace detail {
+
+using json = nlohmann::json;
+
+
+class CompartmentLocation
+{
+    private:
+      std::uint64_t node_id_;
+      std::uint64_t section_index_;
+      double offset_;
+
+      void setNodeId(int64_t node_id) {
+          if (node_id < 0) {
+              throw SonataError(fmt::format("Node id must be non-negative, got {}", node_id));
+          }
+          node_id_ = static_cast<uint64_t>(node_id);
+      }
+        void setSectionIndex(int64_t section_index) {
```

Comment:
strange that the linking isn't complaining about formatting here, I will have to look into that.

Reply:
I don't get why it passed the format CI though.